contracts/Insurance.sol

```diff
@@ -15,122 +14,112 @@ contract Insurance is IInsurance, Ownable, SafetyWithdraw {
     using LibMath for uint256;
     using LibMath for int256;
 
-    int256 public override constant INSURANCE_MUL_FACTOR = 1000000000;
+    int256 public constant override INSURANCE_MUL_FACTOR = 1000000000;
     uint256 public constant SAFE_TOKEN_MULTIPLY = 1e18;
-    address public immutable TCRTokenAddress;
-    IAccount public account;
     ITracerPerpetualsFactory public perpsFactory;
 
-    struct StakePool { 
-        address market;
-        address collateralAsset;
-        uint256 amount; // amount of underlying collateral in pool
-        address token; // tokenized holdings of pool - not necessarily 1 to 1 with underlying
-        mapping(address => uint256) userDebt; // record of user debt to the pool
-        mapping(address => uint256) lastRewardsUpdate;
-    }
+    address public collateralAsset; // Address of collateral asset
+    uint256 public poolAmount; // amount of underlying collateral in pool
+    address public token; // tokenized holdings of pool - not necessarily 1 to 1 with underlying
 
-    // Tracer market => supported or not supported
-    mapping(address => bool) internal supportedTracers;
-    // Tracer market => StakePool
-    mapping(address => StakePool) internal pools;
+    ITracerPerpetualSwaps public tracer; // Tracer associated with Insurance Pool
 
     event InsuranceDeposit(address indexed market, address indexed user, uint256 indexed amount);
     event InsuranceWithdraw(address indexed market, address indexed user, uint256 indexed amount);
     event InsurancePoolDeployed(address indexed market, address indexed asset);
-    event InsurancePoolRewarded(address indexed market, uint256 indexed amount);
 
-    constructor(address TCR) public {
-        TCRTokenAddress = TCR;
+    constructor(address _tracer) {
+        require(perpsFactory.validTracers(_tracer), "Pool not deployed by perpsFactory");
+ 
+        tracer = ITracerPerpetualSwaps(_tracer);
+        InsurancePoolToken _token = new InsurancePoolToken("Tracer Pool Token", "TPT");
+        token = address(_token);
+        collateralAsset = tracer.tracerBaseToken();
+        poolAmount = 0;
+
+        // emit InsurancePoolDeployed(_tracer, tracer.tracerBaseToken());
     }
 
     /**
      * @notice Allows a user to stake to a given tracer market insurance pool
      * @dev Mints amount of the pool token to the user
      * @param amount the amount of tokens to stake
-     * @param market the address of the tracer market to provide insurance
      */
-    function stake(uint256 amount, address market) external override {
-        StakePool storage pool = pools[market];
-        IERC20 token = IERC20(pool.collateralAsset);
-        require(supportedTracers[market], "INS: Tracer not supported");
+    function stake(uint256 amount) external override {
+        require(perpsFactory.validTracers(address(tracer)), "Pool not deployed by perpsFactory");
```

Comment:
Don't think this check is needed since we already validate this in the constructor

Reply:
perpsFactory could be changed at the time of the commit, thus requiring it -- this has been changed so that perpsFactory can't change (only set at construction) so it's already validated and the check isn't required

Reply:
@BenjaminPatch @kumar-ish good point. It should probably also be called on deposit then hey? Otherwise ratios will be out